src/flatty/hashy.nim

```diff
@@ -8,17 +8,24 @@ proc hash*(x: Hash): Hash = x
 
 {.push overflowChecks: off.}
 
+when defined(release):
+  {.push checks: off.}
+
 proc ryan64nim*(p: pointer, len: int): int =
   let
     bytes = cast[ptr UncheckedArray[uint8]](p)
     ints = cast[ptr UncheckedArray[int]](p)
     intSize = sizeof(int)
-  var h: Hash
-  for i in 0 ..< len div intSize:
+  var
+    h: Hash
+    start = 0
+    stop = len div intSize
+  for i in start ..< stop:
     let c = ints[i]
     h = h !& c.hash()
-  let last = (len div intSize) * intSize
-  for i in 0 ..< last + len mod intSize:
+  start = stop * 8
+  stop = start + len mod intSize
```

Comment:
why is the len mod intSIze needed?... can you do `len` ... or `stop*8 ..< len` ?